assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image/frontend.tsx

```diff
@@ -72,7 +72,11 @@ interactivityStore(
 					context: Context;
 					event: Event;
 				} ) => {
-					if ( ( event.target as HTMLElement ).tagName === 'IMG' ) {
+					if (
+						( event.target as HTMLElement ).classList.contains(
+							'wc-block-product-gallery-dialog-on-click'
+						)
```

Comment:
Is there a reason for this change?

Reply:
Yes, the reason for this change is: that the `View All` thumbnails overlay is not an IMG and we still want it to trigger the opening of the Dialog.

Reply:
We should create another action for the View All link. In this way, we can avoid the check on the CSS class.

Reply:
https://github.com/woocommerce/woocommerce-blocks/pull/11087#discussion_r1355718444

Reply:
I feel like there should be a separate action "view all" for the product-gallery-thumbnails because we're in the context of thumbnails and not the large image right? Or am I missing something?

Reply:
To me it's essetially the same context, as both do the same thing: Open the Lightbox (Dialog). No matter about the component that's being clicked.

Reply:
In the future when another developer looks into an issue with the gallery thumbnails, instinctively they will look into the product-gallery-thumbnails section as this pertains to that. It might confuse them, the code lives in the product-gallery-large-image. That's the only concern I had on that.

Aside: How come we need to check for the class onclick? If the action filter is named specific to opening the lightbox, we can just open it when click without first checking for specific class name and we can remove all these https://github.com/woocommerce/woocommerce-blocks/pull/11087/files#diff-865f98c619d1ca4f27d026d7c9bdd132a9d049158ff3b82701ed2e0426b42fc9R52-R53

Reply:
> In the future when another developer looks into an issue with the gallery thumbnails, instinctively they will look into the product-gallery-thumbnails section as this pertains to that. It might confuse them, the code lives in the product-gallery-large-image. 

This is very valid. If this action is supposed to be shared among components it should live on the parent, Product Gallery.

> How come we need to check for the class onclick? If the action filter is named specific to opening the lightbox, we can just open it when click without first checking for specific class name and we can remove all these

So:
- The original action had an `IMG` check so it wouldn't get triggered by clicking on the Next/Previous buttons, Sale badge etc.
- To keep the code DRY I wanted to reuse the same Dialog opening action for the `View All` overlay.
- To achieve that I needed to remove the `IMG` check as the overlay is not an image.
- I have replaced it with a class check for better control and to enable it for both the Large Image and `View All` overlay.